src/util/schema-translator.ts

```diff
@@ -0,0 +1,92 @@
+import { IFrontendEngineData } from "@lifesg/web-frontend-engine/components";
+import { EElementType, TElement, TElementMap } from "src/context-providers";
+
+const emailFieldToSchema = (element: TElement) => {
+    const emailFieldSchema = {
+        [element.id]: {
+            label: element.label,
+            uiType: element.type,
+            columns: {
+                desktop: element.columns,
+            },
+            placeholder: element.placeholder,
+
+            validation: [
+                {
+                    required: element.required,
+                    errorMessage: element.requiredErrorMsg,
+                },
+            ],
+        },
+    };
+
+    return emailFieldSchema;
+};
+
+const textBasedFieldToSchema = (element: TElement) => {
+    const textBasedFieldSchema = {
+        [element.id]: {
+            label: element.label,
+            uiType: element.type,
+            columns: {
+                desktop: element.columns,
+            },
+            placeholder: element.placeholder,
+
+            validation: [
+                {
+                    required: element.required,
+                    errorMessage: element.requiredErrorMsg,
+                },
+            ],
+        },
+    };
+
+    return textBasedFieldSchema;
+};
+
+const generateSchemaByType = (element: TElement) => {
+    switch (element.type) {
+        case EElementType.EMAIL:
+            return emailFieldToSchema(element);
+        case EElementType.TEXT:
+        case EElementType.TEXTAREA:
+        case EElementType.CONTACT:
+        case EElementType.NUMERIC:
+            return textBasedFieldToSchema(element);
+    }
+};
+
+export function generateSchema(elements: TElementMap) {
+    let fields = {};
+
+    Object.entries(elements).forEach(([_, value]) => {
+        const translatedChild = generateSchemaByType(value);
+        fields = { ...fields, ...translatedChild };
+    });
```

Comment:
- use `Object.values()` since we are dealing with the values only
- can we use `.reduce()` instead of `.forEach()` to build the `fields` object?